src/evict/evict_lru.c

```diff
@@ -1303,6 +1303,18 @@ __evict_lru_walk(WT_SESSION_IMPL *session)
         }
     }
 
+    /*
+     * Discard remaining pages by decrementing entries until it matches eviction candidates and
+     * clearing each entry from eviction list.
+     */
+    while (entries > queue->evict_candidates)
+        __evict_list_clear(session, &queue->evict_queue[--entries]);
+
+    queue->evict_entries = entries;
+
+    /* The number of entries should match the number of candidates at this point. */
+    WT_ASSERT(session, queue->evict_entries == queue->evict_candidates);
```

Comment:
You do not need an assert here, as this is also the terminating condition for the while loop and is hence guaranteed to be true.

Reply:
Make sense, removed in [110aa3e](https://github.com/wiredtiger/wiredtiger/pull/8755/commits/110aa3ebfddb8d23f38fde53c705319ba256526d)